eth/peerset.go

```diff
@@ -458,7 +458,12 @@ func (ps *peerSet) registerPeer(peer *eth.Peer, ext *snap.Peer, diffExt *diff.Pe
 		eth.trustExt = &trustPeer{trustExt}
 	}
 	if bscExt != nil {
-		eth.bscExt = &bscPeer{bscExt}
+		if err := bscExt.Handshake(); err == nil {
```

Comment:
Does this seem against the design of `geth`, as if every protocol follow the same pattern, the handshake of the different protocol will happen serially.

Reply:
the design of geth has no handshake exclude eth protocol
waitBscExtension and registerBscExtension can end quickly
when handshake of sub protocol involved in above two , it' begin to slow down and lead to problem